overlord/snapstate/snapmgr.go

```diff
@@ -641,14 +645,18 @@ func snapSetupAndState(t *state.Task) (*SnapSetup, *SnapState, error) {
 
 func (m *SnapManager) undoMountSnap(t *state.Task, _ *tomb.Tomb) error {
 	t.State().Lock()
-	ss, _, err := snapSetupAndState(t)
+	ss, snapst, err := snapSetupAndState(t)
 	t.State().Unlock()
 	if err != nil {
 		return err
 	}
 
 	pb := &TaskProgressAdapter{task: t}
-	return m.backend.UndoSetupSnap(ss.placeInfo(), pb)
+	typ, err := snapst.Type()
```

Comment:
is this case fully tested?  atm the code is setting type very late (doLinkSnap) so I think in some scenarios we will tend to explode there (we need a test failing in CopyData I think),  

we can try to set Type earlier, it was set late to cope with preexisting changes
safest maybe to set it both early and late for a while